### PR TITLE
[EGD-8094] Change mp3 decoder replaced with dr_mp3

### PR DESCRIPTION
**Pull request by kkleczkowski**
Mp3 decoder replaced with dr_mp3.

### CODE REVIEW

**Review by lucckb — APPROVED** [review]
In the dr_mp3 code please use rather malloc/free instead of new /delete because it is rather ANSI C not ++

File: module-apps/application-music-player/ApplicationMusicPlayer.cpp [hunk 1]
```diff
@@ -32,7 +32,7 @@ namespace app
         };
     } // namespace music_player::internal
 
-    constexpr std::size_t applicationMusicPlayerStackSize = 5 * 1024;
+    constexpr std::size_t applicationMusicPlayerStackSize = 8 * 1024;
```
Comment (alekrudnik) [inline module-apps/application-music-player/ApplicationMusicPlayer.cpp:35]: why needed? this is application stack size

File: module-audio/Audio/decoder/decoderMP3.cpp [hunk 2]
```diff
@@ -1,9 +1,9 @@
 // Copyright (c) 2017-2021, Mudita Sp. z.o.o. All rights reserved.
 // For licensing, see https://github.com/mudita/MuditaOS/LICENSE.md
 
-#define MINIMP3_ONLY_MP3
-#define MINIMP3_NO_SIMD
-#define MINIMP3_IMPLEMENTATION
+#define DR_MP3_IMPLEMENTATION
+#define DR_MP3_NO_STDIO
+#define DR_MP3_NO_SIMD
```
Comment (alekrudnik) [inline module-audio/Audio/decoder/decoderMP3.cpp:6]: why no SIMD?

File: module-audio/Audio/decoder/decoderMP3.cpp [hunk 3]
```diff
@@ -20,241 +20,56 @@ namespace audio
             return;
         }
 
-        mp3d = std::make_unique<mp3dec_t>();
+        mp3 = new drmp3;
```
Comment (alekrudnik) [inline module-audio/Audio/decoder/decoderMP3.cpp:23]: is it freed anywhere?

File: module-audio/Audio/decoder/decoderMP3.cpp [hunk 4]
```diff
@@ -20,241 +20,56 @@ namespace audio
...
     {
 
-        mp3dec_frame_info_t info = {};
-
-        int32_t bytesAvailable = DECODER_BUFFER_SIZE;
-        uint32_t bufferIndex   = 0;
-
-        auto decBuffer = std::make_unique<uint8_t[]>(DECODER_BUFFER_SIZE);
+        uint32_t samplesRead = 0;
+        samplesRead          = drmp3_read_pcm_frames_s16(mp3, samplesToRead / chanNumber, (drmp3_int16 *)pcmData);
```
Comment (alekrudnik) [inline module-audio/Audio/decoder/decoderMP3.cpp:54]: reinterpret cast?

File: module-audio/Audio/decoder/decoderMP3.cpp [hunk 5]
```diff
@@ -20,241 +20,56 @@ namespace audio
...
-            if (last_refill && (info.frame_bytes == 0) && (smpl == 0)) {
-                return frames_count;
-            }
-
-            // insufficient data
-            if ((info.frame_bytes == 0) && (smpl == 0)) {
-                goto refill;
-            }
-        }
+        decoderMP3 *userdata = (decoderMP3 *)pUserData;
```
Comment (alekrudnik) [inline module-audio/Audio/decoder/decoderMP3.cpp:65]:
```suggestion
        auto *userdata = reinterpret_cast(decoderMP3 *)pUserData;
```

File: module-audio/Audio/decoder/decoderMP3.cpp [hunk 6]
```diff
@@ -20,241 +20,56 @@ namespace audio
...
-
-        // Manage samples in internal buffer
-        pcmsamplesbuffer_idx = samplesFetched - samplesToReadChann;
-        memmove(&pcmsamplesbuffer[0], &pcmsamplesbuffer[samplesToReadChann], pcmsamplesbuffer_idx * sizeof(int16_t));
-
-        /* Calculate frame duration in seconds */
-        position += (float)((float)(chanNumber == 2 ? samplesToRead / chanNumber : samplesToRead) / (float)sampleRate);
-
-        return samplesToRead;
+        decoderMP3 *userdata = (decoderMP3 *)pUserData;
```
Comment (alekrudnik) [inline module-audio/Audio/decoder/decoderMP3.cpp:71]: same as above

File: module-audio/Audio/decoder/decoderMP3.hpp [hunk 7]
```diff
@@ -32,24 +34,48 @@ namespace audio
...
-
-        uint32_t samplesPerFrame = 0;
-
-        // Variables below are used during decoding procedure
-        uint32_t firstValidFrameByteSize   = 0;
-        uint32_t firstValidFrameFileOffset = 0;
-        bool lastRefill                    = false;
-        bool decoderNotFirstRun            = false;
-        uint32_t bytesAvailable            = 0;
+        drmp3 *mp3;
```
Comment (alekrudnik) [inline module-audio/Audio/decoder/decoderMP3.hpp:37]:
```suggestion
        drmp3 *mp3 = nullptr;
```

File: module-audio/Audio/decoder/decoderMP3.cpp (hunk 3 above)
Comment (kkleczkowski) [reply module-audio/Audio/decoder/decoderMP3.cpp:23]: It's changed to unique_ptr

File: module-apps/application-music-player/ApplicationMusicPlayer.cpp (hunk 1 above)
Comment (kkleczkowski) [reply module-apps/application-music-player/ApplicationMusicPlayer.cpp:35]: done

File: module-audio/Audio/decoder/decoderMP3.cpp (hunk 5 above)
Comment (kkleczkowski) [reply module-audio/Audio/decoder/decoderMP3.cpp]: done

File: module-audio/Audio/decoder/decoderMP3.cpp (hunk 6 above)
Comment (kkleczkowski) [reply module-audio/Audio/decoder/decoderMP3.cpp]: done

File: module-audio/Audio/decoder/decoderMP3.hpp (hunk 7 above)
Comment (kkleczkowski) [reply module-audio/Audio/decoder/decoderMP3.hpp]: It's changed to unique_ptr

**Review by pholat — APPROVED** [review]

**Pull request merged**